redbiom/fetch.py

```diff
@@ -440,24 +455,41 @@ def taxon_descendents(context, taxon, get=None):
     ---------------------
     SMEMBERS <context>:taxonomy-children:<taxon>
     """
+    import redbiom._requests
+
     if get is None:
         import redbiom
-        import redbiom._requests
         config = redbiom.get_config()
         get = redbiom._requests.make_get(config)
+    hmgetter = redbiom._requests.buffered
 
-    to_get = [taxon]
+    to_get = [(None, taxon), ]
     to_keep = set()
     while to_get:
         new_to_get = []
-        for t in to_get:
-            if t.startswith('terminal:'):
-                to_keep.add(t.split(':', 1)[1])
+        for parent, taxon in to_get:
+            if taxon == 'has-terminal':
+                tips = get(context, 'SMEMBERS', 'terminal-of:%s' % parent)
+                to_keep.update(set(tips))
             else:
-                gotten = get(context, 'SMEMBERS', 'taxonomy-children:%s' % t)
-                new_to_get.extend(gotten)
+                gotten = get(context, 'SMEMBERS',
+                             'taxonomy-children:%s' % taxon)
+                new_to_get.extend([(taxon, child) for child in gotten])
         to_get = new_to_get
-    return to_keep
+
+    remapped_bulk = hmgetter(to_keep, None, 'HMGET', context,
+                             get=get, buffer_size=100,
+                             multikey='feature-index-inverted')
+
+    remapped = {name
```

Comment:
is remapped_bulk a list of list/tuples? Not sure why we need the zip ...

Reply:
`remapped_bulk` yields `(list_of_items_queried, list_of_items_obtained)` which are assured to be in index order w.r.t. to each other.